File: frontend/lib/data/models/product.dart
class Product {
  final int id;
  final int categoryId;
  final String name;
  final String shortDesc;
  final String description;
  final double price;
  final String imageUrl;
  final String? model3dUrl;
  final bool isAvailable;

  Product({
    required this.id,
    required this.categoryId,
    required this.name,
    required this.shortDesc,
    required this.description,
    required this.price,
    required this.imageUrl,
    this.model3dUrl,
    required this.isAvailable,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      categoryId: json['category_id'],
      name: json['name'],
      shortDesc: json['short_desc'] ?? '',
      description: json['description'] ?? '',
      price: json['price']?.toDouble() ?? 0.0,
      imageUrl: json['image_url'] ?? '',
      model3dUrl: json['model3d_url'],
      isAvailable: json['is_available'] ?? true,
    );
  }
}

File: backend/internal/models/product.go
package models

import "time"

type Product struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	CategoryID  uint      `json:"category_id" gorm:"not null"`
	Name        string    `json:"name" gorm:"not null"`
	ShortDesc   string    `json:"short_desc"`
	Description string    `json:"description"`
	Price         float64   `json:"price" gorm:"not null"`
	ImageURL      string    `json:"image_url"`
	Model3dUrl    string    `json:"model3d_url"`
	Model3DIosUrl string    `json:"model3d_ios_url"`
	IsAvailable   bool      `json:"is_available" gorm:"default:true"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
